src/cephadm/cephadm

```diff
@@ -6248,6 +6324,64 @@ def command_adopt_prometheus(ctx, daemon_id, fsid):
     update_firewalld(ctx, daemon_type)
 
 
+def command_adopt_loki(ctx, daemon_id, fsid):
+    # type: (CephadmContext, str, str) -> None
+    daemon_type = 'loki'
+    (uid, gid) = extract_uid_gid_monitoring(ctx, daemon_type)
+
+    _stop_and_disable(ctx, 'loki')
+
+    data_dir_dst = make_data_dir(ctx, fsid, daemon_type, daemon_id,
+                                 uid=uid, gid=gid)
+
+    # config
+    config_src = '/etc/loki/loki.yml'
+    config_src = os.path.abspath(ctx.legacy_dir + config_src)
+    config_dst = os.path.join(data_dir_dst, 'etc/loki')
+    makedirs(config_dst, uid, gid, 0o755)
+    copy_files(ctx, [config_src], config_dst, uid=uid, gid=gid)
+
+    # data
+    data_src = '/var/lib/loki'
+    data_src = os.path.abspath(ctx.legacy_dir + data_src)
+    data_dst = os.path.join(data_dir_dst, 'data')
+    copy_tree(ctx, [data_src], data_dst, uid=uid, gid=gid)
+
+    make_var_run(ctx, fsid, uid, gid)
+    c = get_container(ctx, fsid, daemon_type, daemon_id)
+    deploy_daemon(ctx, fsid, daemon_type, daemon_id, c, uid, gid)
+    update_firewalld(ctx, daemon_type)
+
+
+def command_adopt_promtail(ctx, daemon_id, fsid):
+    # type: (CephadmContext, str, str) -> None
+    daemon_type = 'promtail'
+    (uid, gid) = extract_uid_gid_monitoring(ctx, daemon_type)
+
```

Comment:
Same thing here. What's different between these 2 functions? Can we parametrize it? Let's keep in mind that after this change this file is... 8,700 lines long, so let's contribute not to add more code bloat.

BTW, is it really needed to adopt these services, given they were not running on ceph-ansible or other preexisting deployments?

Reply:
Removed these methods